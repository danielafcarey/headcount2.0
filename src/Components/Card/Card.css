ul {
  list-style-type: none;
  list-style-position: outside;
  text-align: center;
  padding: 0;
}

.card {
  height: 350px;
  width: 200px;
  margin: 20px;
  border: 1px solid black;
  box-shadow: 5px 7px 5px grey;
  text-align: center;
  background-color: white;
}

.title {
  font-family: 'Bangers', cursive;
  letter-spacing:3px;
  color: Orange;
  padding-left: 7px;
  padding-right: 7px;
  font-size: x-large;
  text-shadow: 2px 2px 5px black;
}

.list-items {
  font-family: 'Geo', sans-serif;
  font-size: larger;
}

.above {
  color: white;
  border: 1px solid black;
  background-color: black;
}

.below {
  color: black;
}

.selected {
   box-shadow: 0 0 80px purple;
}