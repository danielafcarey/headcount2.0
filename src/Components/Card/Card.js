import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

const Card = ({ section, title, listOfData, addCompareCard, removeCompareCard }) => {
  const listItems = listOfData.map((data, index) => {
    const year = Object.keys(data)[0];
    const dataNum = Object.values(data)[0];
    const listItem = `${year}: ${dataNum}`;
    let dataStyle; 
    dataNum >= 0.5 ? dataStyle = 'above' : dataStyle = 'below';

    return <li className={ dataStyle } key={ `listItem${index}` }>{ listItem }</li>;
  })

  let clickFunc
  if (section === 'compare') {
    clickFunc = () => removeCompareCard(title) 
  } else {
    clickFunc = () => addCompareCard(title)
  }

  return (
    <div className="card" 
          onClick={clickFunc}>
      <h3>{ title }</h3>
      <ul>{ listItems }</ul>
    </div>
  )

}

Card.propTypes = {
  title: PropTypes.string.isRequired, 
  listOfData: PropTypes.array.isRequired
}

export default Card;
