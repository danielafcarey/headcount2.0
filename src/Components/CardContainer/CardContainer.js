import React from 'react';
import Card from '../Card/Card.js';

const CardContainer = ( { repo } ) => {
  const repoKeys = Object.keys(repo)
  console.log(repoKeys)
  const cardList = repoKeys.map(district => {
    let title = district;
    let listOfData = repo[district];

    return <Card title={title}
                 listOfData={listOfData} />

  })

  return (
      <div className="card-container">
        { cardList }
      </div>
    )
}

export default CardContainer; 