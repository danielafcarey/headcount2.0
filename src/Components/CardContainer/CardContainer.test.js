import React from 'react';
import { shallow, mount } from 'enzyme';
import CardContainer from './CardContainer.js';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import Card from '../Card/Card.js';

describe('CardContainer', () => {
  it('should match the snapshot with no cards', () => {
    const cardContainerSnapshot = renderer.create(<CardContainer repo={ [] }
                                           addCompareCard={ jest.fn() }
                                           removeCompareCard={ jest.fn() }
                                         />).JSON;

    expect(cardContainerSnapshot).toMatchSnapshot();
  })

  it('should match the snapshot with cards', () => {
    const mockProp = [{ 'DISTRICT': [{ "2009": 0.986 }],
                      selected: false 
                   }];
    const cardContainerSnapshot = renderer.create(<CardContainer repo={ mockProp }
                                           addCompareCard={ jest.fn() }
                                           removeCompareCard={ jest.fn() }
                                         />).JSON;

    expect(cardContainerSnapshot).toMatchSnapshot();
  }) 

})
