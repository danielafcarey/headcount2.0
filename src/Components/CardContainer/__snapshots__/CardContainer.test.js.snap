// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with cards 1`] = `undefined`;

exports[`CardContainer should match the snapshot with no cards 1`] = `undefined`;
